res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -158,147 +163,123 @@ VCI102.pitch = function(ch, midino, value, status, group) {
     engine.setValue(group, "pitch_adjust", value);
 };
 
-VCI102.fxKnob = [
-    "[EffectRack1_EffectUnit1_Effect1]",
-    "[EffectRack1_EffectUnit2_Effect1]",
-    "[EffectRack1_EffectUnit3_Effect1]",
-    "[EffectRack1_EffectUnit4_Effect1]"
-];
-
-["parameter1", "parameter2", "parameter3"].forEach(function(key) {
-    VCI102[key] = function(ch, midino, value, status, group) {
-        group = VCI102.fxKnob[ch];
-        if (VCI102.shift[ch % 2]) {
-            // link to super1, inverse variants -> none -> direct variants
-            if (engine.getValue(group, key + "_loaded")) {
-                value = Math.round(value / 16) - 4;
-                engine.setValue(group, key + "_link_type", Math.abs(value));
-                engine.setValue(group, key + "_link_inverse", value < 0);
-            }
-        } else {
-            engine.setParameter(group, key, value < 127 ? value / 128 : 1);
-        }
-    };
-});
-
-VCI102.super1 = function(ch, midino, value, status, group) {
-
-    function getKey() {
-        // if any parameters are linked then "super1" else "mix"
-        var i, j, effect;
-        for (i = engine.getValue(group, "num_effects"); i > 0; i--) {
-            effect = group.slice(0, -1) + "_Effect" + i + "]";
-            for (j = engine.getValue(effect, "num_parameters"); j > 0; j--) {
-                if (engine.getValue(effect, "parameter" + j + "_link_type")) {
-                    return "super1";
-                }
-            }
-        }
-        return "mix";
-    }
-
-    engine.setValue(group, getKey(), value < 127 ? value / 128 : 1);
+VCI102.slot = function(group, n) {
+    // convert effectUnit group to effectSlot group
+    return group.slice(0, -1) + "_Effect" + n + "]";
 };
 
-VCI102.prev_chain = function(ch, midino, value, status, group) {
-    if (VCI102.shift[1 - ch % 2]) {
-        // select Effect1 of the EffectUnit if shift of the other Deck
-        if (value) {
-            VCI102.fxKnob[ch] = group.slice(0, -1) + "_Effect1]";
+VCI102.parameter = function(ch, value, group, key) {
+    if (VCI102.shift[ch % 2]) {
+        // link to meta, inverse variants -> none -> direct variants
```

Comment:
I know some people using compact controllers like this for preparing tracks at home. For those (including me) mapping editing controls (link to meta, waveform zoom, beats translate, ...) is useful.